galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/commons/QueryParameters.java

```diff
@@ -127,6 +128,34 @@ public Instant getSingleInstant(String queryParameterName, Instant defaultValue)
 		return returnedValue;
 	}
 
-	
-
+	public Instant getSingleInstantIfParameterNotPresent(String queryParameterName , String parameterToTest) throws InternalServletException {
+		// This function will return null when neither of the two parameters provided are in the query
+		String paramTestValueStr = getSingleString(parameterToTest, null);
+		Instant result = getSingleInstant(queryParameterName, null);
+		if (paramTestValueStr == null && result ==null) {
+			return null;
+		}
+		return result;
+	}
+		
+	public Instant getDefaultFromInstantIfNoQueryIsPresent () throws InternalServletException{
+		// The default for 'from' is now-24 hours. If no query parameters are specified
+		Integer querysize = this.params.size();
+		Instant from = null ;
+		if (querysize >= 0){
+			from = getSingleInstantIfParameterNotPresent("from", "runname");
+			//Check to see if there is no query (i.e. hit the /ras/runs/ endpoint)
+			if (from == null && querysize == 0){
+				Instant fromDefault = Instant.now().minus(24, ChronoUnit.HOURS);
+				from = getSingleInstant("from", fromDefault);
+			}else if (from == null){
+				/*  RULE: Throw exception because a query exists but no from date has been supplied
+				* EXCEPT: When a runname is present in the query
+				*/
+				ServletError error = new ServletError(GAL5010_FROM_DATE_IS_REQUIRED);
+				throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
This test case is already covered in a test as it should not be throwing this error if `?runname={runname}` since the query size should not be 0 and therefore it should also not error as runname exists.

Simply put it acts as an OR gate where you can have a runname or a from date but not neither